### ISSUE TITLE
Error loading swirl

### ISSUE THREAD

**Issue by TimboSid** [swirldev/swirl#459]
I'm very new to R, and I run on Mac OS X 10.7.5. I downloaded and installed R 3.2.1. I downloaded and installed RStudio 00.99.893.

When I attempt to load swirl with library(swirl) I get this error: 

Error : .onAttach failed in attachNamespace() for 'swirl', details: call: stri_c(..., sep = sep, collapse = collapse, ignore_null = TRUE)
error: object 'C_stri_join' not found
In addition: Warning message:
package ‘swirl’ was built under R version 3.2.5
Error: package or namespace load failed for ‘swirl’

**Comment by seankross** [comment]
Try:

```r
install.packages("stringi")
install.packages("swirl")
library(swirl)
swirl()
```

Let me know if that works for you.

**Comment by N3in** [comment]
I have tried what you suggest Sean, and it still won't load. I have looked into other issues, so I will try to post the answers to questions I have seen you ask elsewhere:

library("swirl")
swilr()

> library("swirl")
Error in loadNamespace(j <- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]]) : 
  there is no package called ‘crayon’
Error: package or namespace load failed for ‘swirl’
> swirl()
Error: could not find function "swirl"

R.version.string

> R.version.string
[1] "R version 3.3.1 (2016-06-21)"

sessionInfo()

> sessionInfo()
R version 3.3.1 (2016-06-21)
Platform: i386-w64-mingw32/i386 (32-bit)
Running under: Windows 7 (build 7601) Service Pack 1

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252    LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                           LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] httr_1.2.1     magrittr_1.5   R6_2.1.2       tools_3.3.1    RCurl_1.95-4.8 stringi_1.1.1  stringr_1.0.0 
[8] digest_0.6.9   bitops_1.0-6

"swirl" %in% installed.packages()

> "swirl" %in% installed.packages()
[1] TRUE

I hope this helps. Any assistance provided would be greatly appreciated.

**Comment by seankross** [comment]
@N3in what happens if you install crayon and then try loading swirl?

**Comment by N3in** [comment]
> install.packages("crayon")
Installing package into ‘\\int.grp/share/NA/HOMEDIR/SUM/SheornM/R/win-library/3.3’
(as ‘lib’ is unspecified)
trying URL 'https://cran.rstudio.com/bin/windows/contrib/3.3/crayon_1.3.2.zip'
Content type 'application/zip' length 703760 bytes (687 KB)
downloaded 687 KB

package ‘crayon’ successfully unpacked and MD5 sums checked
Warning in install.packages :
  unable to move temporary installation ‘\\int.grp\share\NA\HOMEDIR\SUM\SheornM\R\win-library\3.3\file1f1465b03a16\crayon’ to ‘\\int.grp\share\NA\HOMEDIR\SUM\SheornM\R\win-library\3.3\crayon’

The downloaded binary packages are in
	C:\Users\sheornm\AppData\Local\Temp\RtmpYJmjEr\downloaded_packages
> library("swirl")
Error in loadNamespace(j <- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]]) : 
  there is no package called ‘crayon’
Error: package or namespace load failed for ‘swirl’
> swirl()
Error: could not find function "swirl"

**Comment by N3in** [comment]
@seankross also, I have ggplots and dplyr installed. I just got R studio, and these are the only packages I have. (including swirl, crayon, and stringi)

**Comment by N3in** [comment]
I figured it out. R studio was trying to place files into the library, but were being put in another location. Moved files around, and now all is good.

Appreciate the help

**Comment by Scott-Schuster** [comment]
I'm running R 3.3.2, Rstudio 1.0.136,  on WIndows 10

I'm basically brand new to R and to anything aside from typing term papers, and performing google searches. 

when I entered library(swirl) into my R console I was receiving this:
Error in loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = vI[[i]]) : 
  there is no package called ‘stringi’
Error: package or namespace load failed for ‘swirl’

As per Seankross's advice above I entered:
install.packages("stringi")
install.packages("swirl")
library(swirl)
swirl()

It worked, so thanks a bunch seahkross. Hopefully anyone else receiving that same error can find this.
